src/activity_handlers.cpp

```diff
@@ -336,12 +358,32 @@ static void butcher_cbm_group( const std::string &group, const tripoint &pos,
     if( check_butcher_cbm( roll ) ) {
         //The CBM works
         const auto spawned = g->m.put_items_from_loc( group, pos, age );
-        for( const auto &it : spawned ) {
+        for( item *it : spawned ) {
+            if( !flags.empty() ) {
+                for( const std::string flg : flags ) {
+                    it->set_flag( flg );
+                }
+            }
+            if( !faults.empty() ) {
+                for( const std::string flt : faults ) {
+                    it->faults.emplace( fault_id( flt ) );
```

Comment:
Why do you take a vector of `std::string` as parameter when you actually need `fault_id`s? Just take a `std::vector<fault_id>` as parameter.